Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs

```diff
@@ -42,156 +41,191 @@ public UserController(IUserRepository userRepository, UserManager<ApplicationUse
+        public async Task<IActionResult> GetUserByIdAsync(string id)
         {
             try
             {
-                var user = await _userRepository.GetById(id);
-                if (user == null)
-                {
-                    return NotFound();
-                }
-                return Ok(user);
+                return Ok(await _userRepository.GetByIdAsync(id));
             }
-            catch (Exception ex)
+            catch (UserNotFound)
             {
-                throw ex;
+                return NotFound();
             }
         }
 
-
         /**
-         * @api {post} api/User/RegisterUser 
+         * @api {post} api/users 
          * @apiVersion 1.0.0
-         * @apiName User
-         * @apiGroup User
+         * @apiName RegisterUserAsync
+         * @apiParam {object} newUser  object
          * @apiParamExample {json} Request-Example:
-         *      
-         *        {
-         *             "FirstName":"ProjectName",
-         *             "LastName":"SlackChannelName",
-         *             "IsActive":"True",
-         *             "JoiningDate":"01-01-2016"
-         *             "SlackUserId":"SlackUserId"
-         *        }      
+         * {
+         *    "Id":"34d1af3d-062f-4bcd-b6f9-b8fd5165e367",
+         *    "FirstName" : "John",
+         *    "Email" : "jone@promactinfo.com",
+         *    "LastName" : "Doe",
+         *    "SlackUserName" :"John",
+         *    "IsActive" : "True",
+         *    "JoiningDate" :"10-02-2016",
+         *    "NumberOfCasualLeave":0,
+         *    "NumberOfSickLeave":0,
+         *    "UniqueName":null,
+         *    "Role":null,
+         *    "UserName": null
+         * }
          * @apiSuccessExample {json} Success-Response:
          * HTTP/1.1 200 OK 
          * {
-         *     "description":"Add User in Application User Table"
+         *     true
+         * }
+         * @apiError InvalidApiRequestException , ArgumentNullException
+         * @apiErrorExample {json} Error-Response:
+         * HTTP/1.1 400 Bad Request
+         * {
+         *   "error": "InvalidApiRequestException"
+         *   "error": "ArgumentNullException"
          * }
          */
         [HttpPost]
-        [Route("add")]
+        [Route("")]
         [Authorize(Roles = "Admin")]
-        public async Task<IActionResult> RegisterUser([FromBody] UserAc newUser)
+        public async Task<IActionResult> RegisterUserAsync([FromBody] UserAc newUser)
         {
             try
             {
+                await _userRepository.GetProjectUserByGroupNameAsync("Slack123h");
                 string createdBy = _userManager.GetUserId(User);
                 if (ModelState.IsValid)
                 {
-                    await _userRepository.AddUser(newUser, createdBy);
+                    await _userRepository.AddUserAsync(newUser, createdBy);
                     return Ok(true);
                 }
                 return Ok(false);
```

Comment:
I have update code, now modelstate is Invalid then it should be return BadRequest